### PR TITLE
Check NATIVE data can be merged with provided merge policy

### PR DESCRIPTION
**Pull request by ahmetmircik**
tests are here: https://github.com/hazelcast/hazelcast-enterprise/pull/1953

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/config/ConfigValidator.java [hunk 1]
```diff
@@ -263,6 +272,31 @@ public static void checkCacheConfig(InMemoryFormat inMemoryFormat, EvictionConfi
...
+            return;
+        }
+
+        try {
+            if (SplitBrainMergePolicy.class.isAssignableFrom(Class.forName(mergePolicy))) {
+                return;
+            }
+
+            String msg = "Split brain recovery is not supported for '%s'."
+                    + " Because its using merge policy `%s` to merge `%s` data."
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/internal/config/ConfigValidator.java]: minor: `'%s', because it's` instead of `'%s'. Because its`?

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/internal/config/ConfigValidator.java]: fixed

**Comment by ahmetmircik** [comment]
Test Result (1 failure / +1)
`classloading.ThreadLeakTest.testThreadLeak`

**Comment by ahmetmircik** [comment]
run-lab-run

**Pull request merged**